tools/xcms_refine/xcms_refine.r

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env Rscript
+
+# ----- LOG FILE -----
+log_file <- file("log.txt", open  =  "wt")
+sink(log_file)
+sink(log_file, type = "output")
+
+
+# ----- PACKAGE -----
+cat("\tSESSION INFO\n")
+
+#Import the different functions
+source_local <- function(fname) {
+  argv <- commandArgs(trailingOnly = FALSE)
+  base_dir <- dirname(substring(argv[grep("--file=", argv)], 8))
+  source(paste(base_dir, fname, sep = "/"))
+}
+source_local("lib.r")
+
+pkgs <- c("xcms", "batch", "RColorBrewer")
+loadAndDisplayPackages(pkgs)
+cat("\n\n");
+
+# ----- ARGUMENTS -----
+cat("\tARGUMENTS INFO\n")
+# interpretation of arguments given in command line as an R list of objects
+args <- parseCommandArgs(evaluate = FALSE)
+write.table(as.matrix(args), col.names = F, quote = F, sep = "\t")
+
+cat("\n\n")
+
+# ----- PROCESSING INFILE -----
+cat("\tARGUMENTS PROCESSING INFO\n")
+
+#saving the specific parameters
+args_method  <- args$method
+args_image   <- args$image
+args_msLevel <- args$msLevel
+param_args <- list()
+
+if (args_method == "CleanPeaks") {
+  param_args$maxPeakwidth <- as.numeric(args$maxPeakwidth)
+  if (is.na(as.numeric(param_args$maxPeakwidth))) stop("\n\nERROR: The maxPeakwidth argument cannot be coerced to a numeric value.")
+} else if (args_method == "FilterIntensity") {
+  param_args$threshold <- as.numeric(args$threshold)
+  if (is.na(as.numeric(param_args$threshold))) stop("\n\nERROR: The threshold argument cannot be coerced to a numeric value.")
+  param_args$nValues <- as.numeric(args$nValues)
+  if (is.na(as.numeric(param_args$nValues))) stop("\n\nERROR: The nValues argument cannot be coerced to a numeric value.")
+  if (as.integer(param_args$nValues) != param_args$nValues) stop("\n\nERROR: The nValues argument is not an integer value.")
+  param_args$value <- args$value
+} else if (args_method == "MergeNeighboringPeaks") {
+  if (is.na(as.numeric(args$expandRt))) stop("\n\nERROR: The expandRt argument cannot be coerced to a numeric value.")
+  if (is.na(as.numeric(args$expandMz))) stop("\n\nERROR: The expandMz argument cannot be coerced to a numeric value.")
+  if (is.na(as.numeric(args$ppm))) stop("\n\nERROR: The ppm argument cannot be coerced to a numeric value.")
+  if (is.na(as.numeric(args$minProp))) stop("\n\nERROR: The minProp argument cannot be coerced to a numeric value.")
+  param_args$expandRt <- args$expandRt
+  param_args$expandMz <- args$expandMz
+  param_args$ppm      <- args$ppm
+  param_args$minProp  <- args$minProp
+}
```

Comment:
Do we really need this since Galaxy will control the type with `type="float"` and `type="integer"`?

Reply:
Wow! That's cool!  I cannot enter bad input (except for negative values...).  So, I will remove the checks.  Thank you.

Reply:
For negative values, you can add some bounders with min and max
https://docs.galaxyproject.org/en/latest/dev/schema.html#integer-and-float